azimuth_capi/operator.py

```diff
@@ -575,6 +585,23 @@ async def on_cluster_resume(instance, name, namespace, **kwargs):
     await save_cluster_status(instance)
 
 
+@model_handler(
+    api.Cluster,
+    kopf.timer,
+    # we don't set idle timeout, as we want to run during changes
+    interval = settings.timer_interval)
+async def check_cluster_timeout(logger, instance, patch, **kwargs):
+    if instance.spec.paused:
+        logger.info("reconciliation is paused - no action taken")
+        return
+
+    # Trigger a timeout check if we are in a transitional state
+    if instance.status.phase in {None, api.ClusterPhase.PENDING,
```

Comment:
`None` shouldn't be possible here - is it a condition you have seen? You should get `UNKNOWN` instead.

Reply:
I believe I saw this in my testing, because this can trigger before the first update completes, because I have not set the idle timer here.